File: LeetCode Practice Problems/62. Unique Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        int path[m][n] = {0};
        for(int i=0;i<m;i++)
            for(int j=0;j<n;j++)
                path[i][j] = 0;
        
        path[0][0] = 1;
        for(int i=0;i<m;i++)
            for(int j=0;j<n;j++)
            {
                if(i+1<m)
                    path[i+1][j]+=path[i][j];
                if(j+1<n)
                    path[i][j+1]+=path[i][j];
            }
        
        return path[m-1][n-1];
    }
};

File: LeetCode/LeetCode Practice Problems/62-Unique Paths.java
class Solution {
    public int uniquePaths(int n, int m) {
        int dp[][] = new int[n][m];
        
        dp[0][0] = 1;
        for(int i=0;i<n;i++) {
            for(int j=0;j<m;j++) {
                if(i == 0 && j == 0) continue;
                if(i == 0 || j == 0) 
                    dp[i][j] = 1;
                else {
                    dp[i][j] = dp[i-1][j] + dp[i][j-1];
                }
            }
        }
        return dp[n-1][m-1];
    }
}
